testplan/common/entity/base.py

```diff
@@ -253,18 +236,17 @@ def stop(self, is_reversed=False):
                 )
                 resource.logger.error(msg)
                 self.stop_exceptions[resource] = msg
+                # Resource status should be STOPPED even it failed to stop
+                resource.status.change(resource.STATUS.STOPPED)
+            else:
+                if resource.cfg.async_start:
```

Comment:
Relay on resource.cfg sounds something that is internal to an Entity. Like an entity can decide it's asyncness not only based on the cfg.async value, would be better to rely on a property of the entity instead of the cfg of the enity. Though this might be a bigger change and itmay go into another PR

Reply:
I saw that there are several places `cfg.async_start` and `cfg.auto_start` occur in our code, not much change needed, so I add 2 properties into `Resource` class, then directly call `resource.async_start` and `resource.auto_start`.